File: blog-backend/src/main/java/com/fonsi13/blogbackend/services/PostService.java
package com.fonsi13.blogbackend.services;

import com.fonsi13.blogbackend.dto.ApiResponse;
import com.fonsi13.blogbackend.dto.PostCreateRequest;
import com.fonsi13.blogbackend.dto.PostResponseDTO;
import com.fonsi13.blogbackend.dto.PostUpdateRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PostService {

    ApiResponse<PostResponseDTO> createPost(PostCreateRequest request, String currentUsername);

    // Devuelve una "Página" de posts, no una lista simple
    ApiResponse<Page<PostResponseDTO>> getAllPosts(Pageable pageable);

    // Leer un solo post por su URL (Slug)
    ApiResponse<PostResponseDTO> getPostBySlug(String slug);

    // Actualizar un post existente
    ApiResponse<PostResponseDTO> updatePost(String postId, PostUpdateRequest request, String currentUsername);

    // Eliminar un post
    ApiResponse<Void> deletePost(String postId, String currentUsername);
}

File: blog-frontend/src/services/PostService.ts
import { type Post, type PaginatedResponse} from "../types/Post.ts";
import instance from "./Api.ts";
import type {ApiResponse} from "../types/Auth.ts";

export const getPosts = async (page: number, size: number) => {

    const response = await instance.get<ApiResponse<PaginatedResponse<Post>>>('/posts', { params: {
        page: page,
        size: size,
        }});

    return response.data.data;
}

export const getPostBySlug = async (slug: string) => {
    const response = await instance.get<ApiResponse<Post>>(`/posts/${slug}`);

    return response.data.data;
}